activejob/lib/active_job/async_job.rb

```diff
@@ -0,0 +1,102 @@
+require 'concurrent'
+require 'thread_safe'
+
+module ActiveJob
+  # == Active Job Async Job
+  #
+  # When enqueueing jobs with Async Job each job will be executed
+  # asynchronously on a +concurrent-ruby+ thread pool. All job data
+  # is retained in memory. Because job data is not saved to a persistent
+  # datastore there is no additional infrastructure needed and most jobs
+  # will process very quickly. The lack of persistence, however, means
+  # that all unprocessed jobs will be lost on application restart. This
+  # makes in-memory queue adapters unsuitable for most production
+  # environments but excellent for development and testing.
+  #
+  # Read more about Concurrent Ruby {here}[https://github.com/ruby-concurrency/concurrent-ruby].
+  #
+  # To use Async Job set the queue_adapter config to +:async+.
+  #
+  #   Rails.application.config.active_job.queue_adapter = :async
+  #
+  # Async Job supports job queues specified with +queue_as+. Queues
+  # are implemented as individual thread pools. Queues will be created
+  # automatically as needed using a default configuration. Should there be a
+  # need to customize the thread pool behind one or more queues, any
```

Comment:
It's not easy to let go when you've spent a lot of hours on something.

This has definitely made me want to look more into concurrent-ruby :heart::heart::heart: